### PR TITLE
Fix possible error that cause double free on destruction @open sesame 05/20 21:09

### PR DESCRIPTION
**Pull request by zhoonit**
There is an occasional error from `unittest_tizen_capi` 
This PR is a possible fix for the problem.

unittests runs more than 200 times without error with this patch.

```bash
[ RUN      ] nntrainer_capi_nnmodel.train_with_file_01_p
training
data_buffer made
data_buffer made done
size: 2
data_buffer clear
data_buffer end
double free or corruption (out)
Aborted (core dumped)
```

**Changes proposed in this PR:**
join data buffer threads first before clearing vectors that are
used in the target function to (possibly) prevent double free

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### CODE REVIEW

**Comment by zhoonit** [comment]
v2 uploaded.

**Changes in this patch**

Fixed unintended deadlock caused by `databuffer::clear` since `databuffer_file::updateData` from `*_thread`  needed `*_running = false` to join. However in v1,  `*_running` was not set `false` before `*_thread.join()` 

TL DR; clearing order should be  
```
train_running = false;
train_thread.join()
```

whereas V1 was
```
train_thread.join()
train_running = false
```

which caused deadlock.
